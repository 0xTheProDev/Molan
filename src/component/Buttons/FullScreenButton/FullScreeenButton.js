// @flow
import React from 'react';
import { Button, Icon } from 'semantic-ui-react';
import { noop } from 'util/common';

import { FullScreenButtonProps } from './FullScreenButton.flow';

export function FullScreenButton(props: FullScreenButtonProps) {
  const {
    isFullScreenMode,
    FullScreenButtonTitle,
    toggleFullScreenMode,
  } = props;

  return (
    <Button
      basic
      floated="right"
      title={FullScreenButtonTitle}
      active={isFullScreenMode}
      onClick={toggleFullScreenMode}
    >
      <Icon name="expand" />
    </Button>
  );
}

FullScreenButton.defaultProps = {
  FullScreenButtonTitle: 'Full Screen',
  toggleFullScreenMode: noop,
};
