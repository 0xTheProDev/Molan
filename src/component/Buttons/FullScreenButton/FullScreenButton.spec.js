import React from 'react';
import { Button } from 'semantic-ui-react';

import { FullScreenButton } from './FullScreeenButton';

describe('Component::FullScreenButton', () => {
  let
    toggleFullScreenModeStub,
    minProps;

  beforeAll(() => {
    toggleFullScreenModeStub = jest.fn();

    minProps = {
      isFullScreenMode: true,
      toggleFullScreenMode: toggleFullScreenModeStub,
    };
  });

  it('should render with base structure', () => {
    const wrapper = shallow(<FullScreenButton {...minProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke callback when Button clicked', () => {
    const wrapper = shallow(<FullScreenButton {...minProps} />);
    wrapper
      .find(Button)
      .simulate('click');

    expect(toggleFullScreenModeStub).toHaveBeenCalled();
  });
});
