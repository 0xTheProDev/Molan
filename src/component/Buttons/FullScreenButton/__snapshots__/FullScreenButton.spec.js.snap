// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component::FullScreenButton should render with base structure 1`] = `
<Button
  active={true}
  as="button"
  basic={true}
  floated="right"
  onClick={[MockFunction]}
  title="Full Screen"
>
  <Icon
    as="i"
    name="expand"
  />
</Button>
`;
